sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala

```diff
@@ -13,14 +13,16 @@ object MetalsPlugin extends AutoPlugin {
 
   val supportedScala2Versions = BuildInfo.supportedScala2Versions.toList
 
-  override lazy val globalSettings: Seq[Def.Setting[_]] = Seq(
-    semanticdbVersion := BuildInfo.semanticdbVersion
-  )
+  val semanticdbVersion = BuildInfo.semanticdbVersion
 
   override lazy val projectSettings: Seq[Def.Setting[_]] = Seq(
     semanticdbCompilerPlugin := {
-      ("org.scalameta" % "semanticdb-scalac" % semanticdbVersion.value)
-        .cross(CrossVersion.full)
+      if (supportedScala2Versions.contains(scalaVersion.value))
```

Comment:
@ckipp01 what about this? I think this might have been the actual problem, we should not set `semanticdbCompilerPlugin` if we don't currently support the given scala version. If a user had the Metals plugin and `semanticdbEnabled := true` then we would have replaced the original without actually needing to do it. So it would break user's builds even though MEtals does not support the given version.

Reply:
Yea, that makes sense 👍🏼 . So if it's not in our supported versions, we just default to what they want. The only problem (and I don't know how to get around) is that for some reason if someone wants to bump the semanticdb version while using a Scala version that we do support, they can't.

Reply:
I think it's fine, we update the semanticdb version often enough. There should be no actual need to do it manually. Besides, if the user modifies the version we cannot know if it actually supports the given Scala version. Metals should be a source of truth here the same as with Bloop.